infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt

```diff
@@ -1072,16 +1072,126 @@ class SwitchLinkingService @Autowired constructor(
 
         switchJointsByLocationTrack.forEach { (locationTrackId, switchJoints) ->
             val (locationTrack, alignment) = locationTrackService.getWithAlignmentOrThrow(DRAFT, locationTrackId)
+
+            val switchJointsWithSlightlyOverlappingSegmentsSnapped = switchJoints.map { switchLinkingJoint ->
+                switchLinkingJoint.copy(
+                    segments = switchLinkingJoint.segments.map { switchLinkingSegment ->
+                        val referencedLayoutSegment = alignment.segments[switchLinkingSegment.segmentIndex]
+
+                        val layoutSegmentHasExistingSwitch = referencedLayoutSegment.switchId != null
+                        val layoutSegmentSwitchReferencesDifferentSwitch =
+                            layoutSegmentHasExistingSwitch && referencedLayoutSegment.switchId != linkingParameters.layoutSwitchId
+
+                        if (layoutSegmentSwitchReferencesDifferentSwitch) {
+                            tryToSnapOverlappingSwitchSegmentToNearbySegment(alignment, switchLinkingSegment)
+                        } else {
+                            switchLinkingSegment
```

Comment:
Tämähän iffittely ei sinänsä ole välttämätön (snäppäysfunktion voisi ajaa kaikille `switchLinkingSegment`), mutta sanoisin iffittelyn helpottavan logiikan tulkintaa. 

Jos snäppäystä ei tarvita tai sitä ei voida tehdä, niin `tryToSnapOverlappingSwitchSegmentToNearbySegment` palauttaa siis alkuperäisen `switchLinkingSegment`in.

Reply:
Mielestäni tämä iffittely on ihan selkeä ja se muistuttaa siitä ikävähköstä realiteetista että tämä snäppäyslogiikka on ylipäätään tarpeellinen